File: src/test/kotlin/trustline/config/security/GoogleAuthProvider.java
package trustline.config.security;

import lombok.AllArgsConstructor;
import org.springframework.security.authentication.BadCredentialsException;
import org.springframework.stereotype.Component;
import trustline.appuser.model.User;
import trustline.appuser.repository.UserRepository;

@Component
@AllArgsConstructor
public class GoogleAuthProvider {

    private final UserRepository userRepository;
    private final JWTConfig jwtConfig;

    public String authenticateWithGoogle(String googleId) {
        User user = userRepository.findByGoogleId(googleId)
                .orElseThrow(() -> new BadCredentialsException("Invalid Google ID"));
        return jwtConfig.generateToken(user);
    }
}

File: src/main/kotlin/trustline/config/security/GoogleAuthProvider.kt
package trustline.config.security;

import org.springframework.security.authentication.BadCredentialsException
import org.springframework.stereotype.Component
import trustline.appuser.repository.UserRepository

@Component
class GoogleAuthProvider(
    private val userRepository: UserRepository,
    private val jwtConfig: JWTConfigService
) {

    fun authenticateWithGoogle(googleId: String): String {
        val user = userRepository.findByGoogleId(googleId)
            .orElseThrow { BadCredentialsException("Invalid Google ID") }

        val roleNames = userRepository.findRoleNamesByUserId(user.id!!)
        return jwtConfig.generateToken(user, roleNames)
    }
}

